c++/src/kj/compat/http.c++

```diff
@@ -5282,12 +5735,56 @@ private:
       return sendWebSocketError("Missing Sec-WebSocket-Key");
     }
 
+    kj::Maybe<CompressionParameters> acceptedParameters;
+    kj::String agreedParameters;
+    KJ_IF_MAYBE(value, requestHeaders.get(HttpHeaderId::SEC_WEBSOCKET_EXTENSIONS)) {
+
+      kj::StringPtr permittedExtension = "permessage-deflate"_kj;
+      auto extensions = _::splitPartsWithCondition(*value, ',', permittedExtension);
```

Comment:
More string twiddling, should reuse a common `parseWebSocketExtensions()` function.

Reply:
The tricky thing is that the way the server should interpret and accept/decline offers is different from how the client should interpret and accept/decline an agreement. They can share _some_ code but having them share a lot might make it more difficult to follow.

Helper functions + moving stuff on server side: https://github.com/capnproto/capnproto/pull/1480/commits/f29cbd448330fe515f6617bc9c9be69b85c1a351
Another helper + moving stuff on client side: https://github.com/capnproto/capnproto/pull/1480/commits/5dbccbcde5b7daebc5a03c57b642b9a2b9dd22d6
Use helper functions in `retainClientCompressorConfig`: https://github.com/capnproto/capnproto/pull/1480/commits/14168b64a6ce0e4f9d1b03eeaab6567d27a1fccc

`TODO:` One thing I want to come back to is `isValidExtensionName()`, since it checks if the extension `startsWith("permessage-deflate")` even though every time we call it (except for [here](https://github.com/capnproto/capnproto/pull/1480/commits/5dbccbcde5b7daebc5a03c57b642b9a2b9dd22d6#diff-09d28b46bd98866f20dd7241e02c70338a76fdb59a483ed02d4c16bbe6bcc3a1R3908)) we already know it does (we passed a condition to `splitPartsWithCondition()`). I've got to run atm but so I'll figure it out tomorrow morning.